docs/user/build-customization.rst

```diff
@@ -0,0 +1,155 @@
+Build customization
+===================
+
+Read the Docs has a :ref:`well-defined build process <builds>` that may not be enough for all the use cases.
+This page explains how to customize Read the Docs build process to get the most of it and support custom build processes.
```

Comment:
```suggestion
This page explains how to customize the build process to get the most of it and support custom build steps.
```

Another work for `jobs` or `steps` here. We really need to tighten this up, so users actually know what we're talking about. 

Is it:

* Build process: Total series of steps we take to build a set of content (wording gets way trickier here when we're doing non-docs projects...)
* Build steps: Each step of the process, specifically the ones that we run? 
* Build jobs: Non-standard steps that users run? 

@agjohnson thoughts here?

Reply:
So, that sort of matches how I would describe it, but I also feel like it's a little inaccurate :upside_down_face: 

There are discrete `pre_install` and `post_install` jobs, and I see a future with an `install` job too. So I suppose we'd call those "jobs", and "steps" are the collection of those "jobs".

So, in :snake: :

```python
BuildProcess(steps=[
    InstallStep(jobs=[
        PreInstallJob(commands=[]),
        InstallJob(commands=[]),
        PostInstallJob(commands=[]),
    ]),
    BuildStep(jobs=[
        PreBuildJob(commands=[]),
        BuildJob(commands=[]),
        PostBuildJob(commands=[]),
    ]),
    ...
])
```

Perhaps pre/post install jobs are currently "custom jobs" or "user-defined jobs"? Or `build.jobs.install` is a "pre-defined job".

Reply:
> "custom jobs" or "user-defined jobs"? Or build.jobs.install is a "pre-defined job"

This is what I had in mind as well. Maybe I got confusing while writing the document 😅 , but I think it's what I wanted to describe: pre-defined jobs and user-defined jobs.

Reply:
I'm accepting Anthony's suggestion for now, but we can update it later if we find better wording.

Reply:
I really like Anthony's code snippet. I'd argue that could be useful to put in the docs, but maybe too codey?